homeassistant/util/ssl.py

```diff
@@ -64,6 +67,7 @@ class SSLCipherList(StrEnum):
 @cache
 def create_no_verify_ssl_context(
     ssl_cipher_list: SSLCipherList = SSLCipherList.PYTHON_DEFAULT,
+    **kwargs: dict[str, Any],
```

Comment:
This could become an issue

https://docs.python.org/3/library/functools.html#functools.lru_cache

> Distinct argument patterns may be considered to be distinct calls with separate cache entries. For example, f(a=1, b=2) and f(b=2, a=1) differ in their keyword argument order and may have two separate cache entries.

We may need to make this an inner function and sort the kwargs

Reply:
I should better describe that

rename `create_no_verify_ssl_context` to `_create_no_verify_ssl_context`, make `create_no_verify_ssl_context` a wrapper around it and sort the kwargs before passing it to `_create_no_verify_ssl_context` so the cache always gets a sorted kwargs